src/collective/clamav/validator.py

```diff
@@ -47,23 +44,31 @@ def __init__(self, name):
         self.name = name
 
     def __call__(self, value, *args, **kwargs):
-        if hasattr(value, 'seek'):
-            # when submitted a new file 'value' is a
+        if hasattr(value, 'getBlob'):
+            # the value can be a plone.app.blob.field.BlobWrapper
+            # in which case we open the blob file to provide a file interface
+            # as used for FileUpload
+            file_value = value.getBlob().open()
```

Comment:
I think the blob wrapper has a shorthand for `value.getBlob().open().seek(0).read()` --> `value.data`. But I'm not 100% sure and did not check it.

Reply:
You are correct it does, but I was trying to leave the door open for passing streams rather than strings to clamd. (but agree that at the moment there's no difference)

This is the comment from blobwrapper.data :-)

```
    def __str__(self):
        """ return data as a string;  this is highly inefficient as it
            loads the complete blob content into memory, but the method
            is unfortunately still used here and there... """
        return openBlob(self.blob).read()

    data = ComputedAttribute(__str__, 0)
```

Reply:
will deal with this in next PR